DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -66,15 +67,20 @@ class DMObjectBuilder {
 
             _currentObject = DMObjectTree.GetDMObject(varDefinition.ObjectPath);
 
-            if (varDefinition.IsGlobal) {
+            if (varDefinition.IsGlobal || varDefinition.IsToplevel) {
                 variable = _currentObject.CreateGlobalVariable(varDefinition.Type, varDefinition.Name);
             } else {
                 variable = new DMVariable(varDefinition.Type, varDefinition.Name, varDefinition.IsGlobal);
                 _currentObject.Variables[variable.Name] = variable;
             }
 
             try {
-                SetVariableValue(variable, varDefinition.Value, varDefinition.ValType);
+                if (varDefinition.ObjectPath.PathString == "/" && varDefinition.Name == "world") {
```

Comment:
If this case isn't skipped you will end up with null exceptions when accessing the world object at runtime. I think it has to do with the order the runtime initializes variables. I'm not sure why this issue didn't exist before, and there may be other cases like this but I haven't looked into the runtime enough to straighten any of it out if there are.

Reply:
The runtime sets `world` before running the global init proc. Because of your changes to `SetVariableValue`, the global init proc was setting `world` back to null. Since those changes are now removed this special handling is no longer needed.